### PR TITLE
Debugger NPE patch

### PR DESCRIPTION
**Pull request by mlvdv**
Remove an ad hoc mechanism for hiding selected frames in stack tracing that also contained an NPE vulnerability (which @mickjordan reported hitting in FastR).

### CODE REVIEW

**Comment by chrisseaton** [comment]
Just checking: do you know why there are now frames with `null` `CallNodes`? It's because these come from frames called via `callTarget.call()`, rather than using a `CallNode`, and previously these weren't included in `iterateFrames` but they are now.

File: truffle/com.oracle.truffle.sl.tools/src/com/oracle/truffle/sl/tools/debug/SLREPL.java [hunk 1]
```diff
@@ -49,6 +49,8 @@
 
     public static void main(String[] args) {
 
+        System.setProperty("truffle.debug.trace", "false");
+        System.setProperty("truffle.instrumentation.trace", "false");
```
Comment (chumer) [inline truffle/com.oracle.truffle.sl.tools/src/com/oracle/truffle/sl/tools/debug/SLREPL.java]: Useless code.

File: truffle/com.oracle.truffle.api.debug/src/com/oracle/truffle/api/debug/Debugger.java [hunk 2]
```diff
@@ -957,15 +957,9 @@ public FrameInstance visitFrame(FrameInstance frameInstance) {
                     if (stackIndex < contextStackDepth) {
                         final Node callNode = frameInstance.getCallNode();
                         if (callNode != null) {
```
Comment (chumer) [inline truffle/com.oracle.truffle.api.debug/src/com/oracle/truffle/api/debug/Debugger.java]: I'd say this check is now useless. 
You can push frames.add(frameInstance) just before stackIndex++.

File: truffle/com.oracle.truffle.sl.tools/src/com/oracle/truffle/sl/tools/debug/SLREPL.java (hunk 1 above)
Comment (mlvdv) [inline truffle/com.oracle.truffle.sl.tools/src/com/oracle/truffle/sl/tools/debug/SLREPL.java]: fixed

File: truffle/com.oracle.truffle.api.debug/src/com/oracle/truffle/api/debug/Debugger.java (hunk 2 above)
Comment (mlvdv) [inline truffle/com.oracle.truffle.api.debug/src/com/oracle/truffle/api/debug/Debugger.java]: Please explain why useless:  I'm seeing a frame with null call node.

Comment (mlvdv) [inline truffle/com.oracle.truffle.api.debug/src/com/oracle/truffle/api/debug/Debugger.java]: It seems that changes in frame iteration have led to this issue.  Is it now stable?

Comment (chrisseaton) [inline truffle/com.oracle.truffle.api.debug/src/com/oracle/truffle/api/debug/Debugger.java]: Yes I think so. If your code works with the latest Truffle that should be the behaviour from now on. It was a bug that you weren't being given these frames with `null` `CallTarget`s before.

Comment (chumer) [inline truffle/com.oracle.truffle.api.debug/src/com/oracle/truffle/api/debug/Debugger.java]: The code could be simpler. I've simplified it myself.

**Comment by chumer** [comment]
Will integrate as soon as its ready.

**Pull request merged**